openedx/core/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -61,6 +88,16 @@ def save_bearer_token(self, token, request, *args, **kwargs):
 
         super(EdxOAuth2Validator, self).save_bearer_token(token, request, *args, **kwargs)
 
+        is_application_restricted = RestrictedApplication.objects.filter(application=request.client).exists()
+
+        if is_application_restricted:
+            # For now, since Open edX doesn't have the protection of scopes, any access token
+            # that was generated for 'restricted applications' (aka external 3rd parties)
+            # must be automatically expired. So let's update
+            # the token dictionary, so set the expires_in field be
+            # on Jan. 1, 1970
+            token['expires_in'] = -int(time.time())
```

Comment:
This is because the underlying DOT library does not re-read the AccessToken after the presave, so the DOT library returns an 'expires_in' which does not reflect the override. We decided not to fork the underlying DOT library, so we need to include this seemingly redundant code